operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/BulkDependentResource.java

```diff
@@ -30,6 +32,21 @@
    */
   void deleteBulkResourceWithIndex(P primary, R resource, int i, Context<P> context);
 
-  ResourceDiscriminator<R, P> getResourceDiscriminator(int index);
+  /**
+   * Determines whether the specified secondary resource matches the desired state with target index
+   * of a bulk resource as defined from the specified primary resource, given the specified
+   * {@link Context}.
+   *
+   * @param actualResource the resource we want to determine whether it's matching the desired state
+   * @param primary the primary resource from which the desired state is inferred
+   * @param context the context in which the resource is being matched
+   * @return a {@link Result} encapsulating whether the resource matched its desired state and this
+   *         associated state if it was computed as part of the matching process. Use the static
+   *         convenience methods ({@link Result#nonComputed(boolean)} and
+   *         {@link Result#computed(boolean, Object)})
+   */
+  Result<R> match(R actualResource, P primary, int index, Context<P> context);
+
+  Optional<R> getSecondaryResource(P primary, int index, Context<P> context);
```

Comment:
So maybe this should be `List<R> getSecondryResources(P primary, Context<P> context)`